### PR TITLE
fix(DB/Item) Flamewrought Key zhCN translation

### PR DESCRIPTION
**Pull request by 55Honey**
## Changes Proposed:
- `Flamewrought Key` gets a zhCN translation

## Issues Addressed:
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/4364

## Tests Performed:
- none

## How to Test the Changes:
- Apply the SQL changes and run the core with a zhCN client. 
- .additem 30622 for Alliance
- .additem 30637 for Horde
- make sure the items have the proper translation

## Known Issues and TODO List:

- [x] Can someone confirm this is actually the right translation?

## Target Branch(es):
- [x] Master

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

**Comment by 55Honey** [comment]
image
Seems good to me.

File: data/sql/base/db_world/item_template_locale.sql [hunk 1]
```diff
@@ -168965,6 +168966,7 @@ INSERT INTO `item_template_locale` VALUES
 (30635,'deDE','Schlüssel der Zeit','Eröffnet den Schwierigkeitsgrad \'Heroisch\' für Dungeons der Höhlen der Zeit.',18019),
 (30636,'deDE','NPC Equip 30636','',15050),
 (30637,'deDE','Flammengeschmiedeter Schlüssel','Eröffnet den Schwierigkeitsgrad \'Heroisch\' für Dungeons der Höllenfeuerzitadelle.',18019),
+(30637,'zhCN', '焰铸钥匙', '允许你进入英雄难度的地狱火堡垒地下城。', 15050),
```
Comment (FrancescoBorzi) [inline data/sql/base/db_world/item_template_locale.sql:168969]: hi, thanks for your contributions. However, we don't modify the base files directly like this, please read this how-to:

https://www.azerothcore.org/wiki/Dealing-with-SQL-files

**Review by FrancescoBorzi — CHANGES_REQUESTED** [review]

File: data/sql/base/db_world/item_template_locale.sql (hunk 1 above)
Comment (55Honey) [reply data/sql/base/db_world/item_template_locale.sql]: I tried, should be good now.

File: data/sql/updates/pending_db_world/rev_1611865917828980400.sql [hunk 2]
```diff
@@ -0,0 +1,3 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1611865917828980400');
+INSERT INTO item_template_locale (ID, locale, Name, Description, VerifiedBuild) VALUES ('30622', 'zhCN', '焰铸钥匙', '允许你进入英雄难度的地狱火堡垒地下城。', '15050'); 
+INSERT INTO item_template_locale (ID, locale, Name, Description, VerifiedBuild) VALUES ('30637', 'zhCN', '焰铸钥匙', '允许你进入英雄难度的地狱火堡垒地下城。', '15050');
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1611865917828980400.sql:3]: Nice, this is better, now I will teach you something =)

1. Always use DELETE before INSERT
2. You can use 1 INSERT here
3. always use \` (backticks) around SQL keywords
4. Never use \'\' around numbers
5. and add a newline ;)

Comment (55Honey) [reply data/sql/updates/pending_db_world/rev_1611865917828980400.sql]: I tried ;)

File: data/sql/updates/pending_db_world/rev_1611865917828980400.sql [hunk 3]
```diff
@@ -0,0 +1,6 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1611865917828980400');
+DELETE FROM `item_template_locale` WHERE `ID` = 30622 and locale = 'zhCN'
+DELETE FROM `item_template_locale` WHERE `ID` = 30637 and locale = 'zhCN'
+INSERT INTO `item_template_locale` (`ID`, `locale`, `Name`, `Description`, `VerifiedBuild`) VALUES
+(30622, 'zhCN', '焰铸钥匙', '允许你进入英雄难度的地狱火堡垒地下城。', 15050),
+(30637, 'zhCN', '焰铸钥匙', '允许你进入英雄难度的地狱火堡垒地下城。', 15050);
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1611865917828980400.sql:6]:
```suggestion
DELETE FROM `item_template_locale` WHERE `ID` IN (30622,30637) and locale = 'zhCN';
INSERT INTO `item_template_locale` (`ID`, `locale`, `Name`, `Description`, `VerifiedBuild`) VALUES
(30622, 'zhCN', '焰铸钥匙', '允许你进入英雄难度的地狱火堡垒地下城。', 15050),
(30637, 'zhCN', '焰铸钥匙', '允许你进入英雄难度的地狱火堡垒地下城。', 15050);
```

Comment (Kitzunu) [reply data/sql/updates/pending_db_world/rev_1611865917828980400.sql:6]: Almost ;)

Comment (55Honey) [reply data/sql/updates/pending_db_world/rev_1611865917828980400.sql:6]: Trying to remember that 👍

**Comment by 55Honey** [comment]
I've found one more. That should be it.

File: data/sql/updates/pending_db_world/rev_1611865917828980400.sql [hunk 4]
```diff
@@ -1,3 +1,6 @@
 INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1611865917828980400');
-INSERT INTO item_template_locale (ID, locale, Name, Description, VerifiedBuild) VALUES ('30622', 'zhCN', '焰铸钥匙', '允许你进入英雄难度的地狱火堡垒地下城。', '15050'); 
-INSERT INTO item_template_locale (ID, locale, Name, Description, VerifiedBuild) VALUES ('30637', 'zhCN', '焰铸钥匙', '允许你进入英雄难度的地狱火堡垒地下城。', '15050');
\ No newline at end of file
+DELETE FROM `item_template_locale` WHERE `ID` = 30622 and locale = 'zhCN'
+DELETE FROM `item_template_locale` WHERE `ID` = 30637 and locale = 'zhCN'
```
Comment (FrancescoBorzi) [inline data/sql/updates/pending_db_world/rev_1611865917828980400.sql]: can be just one `DELETE`:

```sql
DELETE FROM `item_template_locale` WHERE locale = 'zhCN' AND `ID` IN (30622, 30637);
```

the rest looks good, well done!

Comment (Kitzunu) [reply data/sql/updates/pending_db_world/rev_1611865917828980400.sql]: this was fixed several commits ago xD

**Review by Kitzunu — APPROVED** [review]

File: data/sql/updates/pending_db_world/rev_1611865917828980400.sql (hunk 4 above)
Comment (55Honey) [reply data/sql/updates/pending_db_world/rev_1611865917828980400.sql]: should be done already in the last commit. and thanks :)

**Comment by FrancescoBorzi** [comment]
good job @55Honey !

**Comment by pklloveyou** [comment]
Tested

**Comment by Kitzunu** [comment]
Thanks

**Comment by luolingjun** [comment]
INSERT INTO `gameobject_template_locale` (`entry`, `locale`, `name`, `castBarCaption`, `VerifiedBuild`) VALUES ('37', 'zhCN', '伊莉莎的墓碑', '', '18019');
INSERT INTO `item_template_locale` (`ID`, `locale`, `Name`, `Description`, `VerifiedBuild`) VALUES ('30622', 'zhCN', '焰铸钥匙', '允许你进入英雄难度的地狱火堡垒地下城。', '15050');
INSERT INTO `item_template_locale` (`ID`, `locale`, `Name`, `Description`, `VerifiedBuild`) VALUES ('30623', 'zhCN', '水库钥匙', '允许你进入英雄难度的盘牙水库地下城。', '15050');
INSERT INTO `item_template_locale` (`ID`, `locale`, `Name`, `Description`, `VerifiedBuild`) VALUES ('30633', 'zhCN', '奥金尼钥匙', '允许你进入英雄难度的奥金顿地下城。', '15050');
INSERT INTO `item_template_locale` (`ID`, `locale`, `Name`, `Description`, `VerifiedBuild`) VALUES ('30634', 'zhCN', '星船钥匙', '允许你进入英雄难度的风暴要塞地下城。', '15050');
INSERT INTO `item_template_locale` (`ID`, `locale`, `Name`, `Description`, `VerifiedBuild`) VALUES ('30635', 'zhCN', '时光之钥', '允许你进入英雄难度的时光之穴地下城。', '15050');
INSERT INTO `item_template_locale` (`ID`, `locale`, `Name`, `Description`, `VerifiedBuild`) VALUES ('30637', 'zhCN', '焰铸钥匙', '允许你进入英雄难度的地狱火堡垒地下城。', '15050');

**Comment by luolingjun** [comment]
INSERT INTO `gameobject_template_locale` (`entry`, `locale`, `name`, `castBarCaption`, `VerifiedBuild`) VALUES ('37', 'zhCN', '伊莉莎的墓碑', '', '18019');
INSERT INTO `item_template_locale` (`ID`, `locale`, `Name`, `Description`, `VerifiedBuild`) VALUES ('30622', 'zhCN', '焰铸钥匙', '允许你进入英雄难度的地狱火堡垒地下城。', '15050');
INSERT INTO `item_template_locale` (`ID`, `locale`, `Name`, `Description`, `VerifiedBuild`) VALUES ('30623', 'zhCN', '水库钥匙', '允许你进入英雄难度的盘牙水库地下城。', '15050');
INSERT INTO `item_template_locale` (`ID`, `locale`, `Name`, `Description`, `VerifiedBuild`) VALUES ('30633', 'zhCN', '奥金尼钥匙', '允许你进入英雄难度的奥金顿地下城。', '15050');
INSERT INTO `item_template_locale` (`ID`, `locale`, `Name`, `Description`, `VerifiedBuild`) VALUES ('30634', 'zhCN', '星船钥匙', '允许你进入英雄难度的风暴要塞地下城。', '15050');
INSERT INTO `item_template_locale` (`ID`, `locale`, `Name`, `Description`, `VerifiedBuild`) VALUES ('30635', 'zhCN', '时光之钥', '允许你进入英雄难度的时光之穴地下城。', '15050');
INSERT INTO `item_template_locale` (`ID`, `locale`, `Name`, `Description`, `VerifiedBuild`) VALUES ('30637', 'zhCN', '焰铸钥匙', '允许你进入英雄难度的地狱火堡垒地下城。', '15050');

**Comment by FrancescoBorzi** [comment]
@luolingjun please [open a new PR](https://www.azerothcore.org/wiki/How-to-create-a-PR) to submit more code, thank you!

**Pull request merged**